src/util/psbt/serialize.rs

```diff
@@ -71,12 +76,36 @@ impl Serialize for PublicKey {
 }
 
 impl Deserialize for PublicKey {
+    #[inline]
     fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
         PublicKey::from_slice(bytes)
             .map_err(|_| encode::Error::ParseFailed("invalid public key"))
     }
 }
 
+impl Serialize for PartialSignature {
+    fn serialize(&self) -> Vec<u8> {
+        let mut buf = self.signature.serialize_der().to_vec();
+        buf.push(self.sighash_type as u8);
+        buf
+    }
+}
+
+impl Deserialize for PartialSignature {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        let (sig_slice, sighash_byte) = match bytes.len() {
+            70..=73 => (&bytes[..bytes.len() - 1], bytes[bytes.len() - 1]),
+            _ => return Err(encode::Error::ParseFailed("non-standard partial signature length"))
```

Comment:
Is there any standardness rule around sig size? In any case, this should allow for non-standard things